src/main/java/fi/hsl/transitdata/metroats/MetroEstimatesFactory.java

```diff
@@ -0,0 +1,201 @@
+        if (!maybeMetroTrainTypeAts.isPresent()) {
+            return Optional.empty();
+        }
+        metroEstimateBuilder.setTrainType(maybeMetroTrainTypeAts.get());
+        // journeySectionprogress
+        Optional<MetroAtsProtos.MetroProgress> maybeMetroAtsProgress = getMetroAtsProgress(metroEstimate.journeySectionprogress);
+        if (!maybeMetroAtsProgress.isPresent()) {
+            return Optional.empty();
+        }
+        metroEstimateBuilder.setJourneySectionprogress(maybeMetroAtsProgress.get());
+        metroEstimateBuilder.setBeginTime(metroEstimate.beginTime);
+        metroEstimateBuilder.setEndTime(metroEstimate.endTime);
+
+        // Set fields from Redis into metroEstimateBuilder
+        Optional<Map<String, String>> metroJourneyData = getMetroJourneyData(metroId);
+        if(!metroJourneyData.isPresent()) {
+            log.warn("Couldn't read metroJourneyData from redis. Metro ID: {}, redis map: {}.", metroId, metroJourneyData);
+            return Optional.empty();
+        }
+        metroJourneyData.ifPresent(map -> {
+            if (map.containsKey(TransitdataProperties.KEY_START_DATETIME))
+                metroEstimateBuilder.setStartDatetime(map.get(TransitdataProperties.KEY_START_DATETIME));
+            if (map.containsKey(TransitdataProperties.KEY_DIRECTION))
+                metroEstimateBuilder.setDirection(map.get(TransitdataProperties.KEY_DIRECTION));
+            if (map.containsKey(TransitdataProperties.KEY_START_STOP_NUMBER))
+                metroEstimateBuilder.setStartStopNumber(map.get(TransitdataProperties.KEY_START_STOP_NUMBER));
+            if (map.containsKey(TransitdataProperties.KEY_START_STOP_SHORT_NAME))
+                metroEstimateBuilder.setStartStopShortName(map.get(TransitdataProperties.KEY_START_STOP_SHORT_NAME));
+            if (map.containsKey(TransitdataProperties.KEY_OPERATING_DAY))
+                metroEstimateBuilder.setOperatingDay(map.get(TransitdataProperties.KEY_OPERATING_DAY));
+            if (map.containsKey(TransitdataProperties.KEY_START_TIME))
+                metroEstimateBuilder.setStartTime(map.get(TransitdataProperties.KEY_START_TIME));
+            if (map.containsKey(TransitdataProperties.KEY_ROUTE_NAME))
+                metroEstimateBuilder.setRouteName(map.get(TransitdataProperties.KEY_ROUTE_NAME));
+        });
+
+        // routeRows
+        List<MetroAtsProtos.MetroStopEstimate> metroStopEstimates = new ArrayList<>();
+        for (MetroStopEstimate metroStopEstimate : metroEstimate.routeRows) {
+            Optional<MetroAtsProtos.MetroStopEstimate> maybeMetroStopEstimate = toMetroStopEstimate(metroStopEstimate);
+            if (!maybeMetroStopEstimate.isPresent()) {
+                    return Optional.empty();
+            } else {
+                metroStopEstimates.add(maybeMetroStopEstimate.get());
+            }
+        };
+        metroEstimateBuilder.addAllMetroRow(metroStopEstimates);
```

Comment:
`MetroRows`instead of `MetroRow` in protobuf might be better.

Reply:
Changed now:

message MetroEstimate {
repeated MetroStopEstimate metro_row = 2;
-->
repeated MetroStopEstimate metro_rows = 2;